src/coreclr/jit/gentree.cpp

```diff
@@ -31691,6 +31691,7 @@ GenTree* Compiler::gtFoldExprHWIntrinsic(GenTreeHWIntrinsic* tree)
                         // op3 has side effects, this would require us to append a new statement
                         // to ensure that it isn't lost, which isn't safe to do from the general
                         // purpose handler here. We'll recognize this and mark it in VN instead
+                        break;
```

Comment:
The `break;` here was apparently lost somewhere and so it was incorrectly folding down to `op2` and losing the side effect